corehq/apps/reports/static/reports/v2/js/datagrid/binding_handlers.js

```diff
@@ -15,18 +15,87 @@ hqDefine('reports/v2/js/datagrid/binding_handlers', [
 ) {
     'use strict';
 
+    var _select2Init = function ($select, options, params) {
+        var initialValue;
+
+        $select.select2(params);
+
+        if (!_.isFunction(options.getInitialValue)) return;
+
+        initialValue = options.getInitialValue();
+
+        if (options.multiple) {
+            // initializing multi select2s
+            if (!_.isObject(initialValue) && !_.isArray(initialValue)) {
+                initialValue = [{text: initialValue, id: initialValue}];
+            }
+
+            if (!_.isArray(initialValue)) {
+                initialValue = [initialValue];
+            }
+
+            if (options.url) {
+                // only hard load options when async select2 is being used
+                _.each(initialValue, function (valObj) {
+                    var option = new Option(valObj.text, valObj.id, true, true);
+                    $select.append(option);
+                });
+            }
+            $select.trigger('change');
+            $select.trigger({type: 'select2:select', params: {data: initialValue}});
+
+        } else if (!_.isArray(initialValue)) {
```

Comment:
Nitpick: this condition doesn't seem necessary (or like it should ever happen for a single select).